pkg/clustermesh/namespace/mock_namespace_manager.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package namespace
+
+import (
+	"context"
+	"testing"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/hive/hivetest"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	k8sFakeClient "github.com/cilium/cilium/pkg/k8s/client/testutils"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+)
+
+// NewMockNamespaceManager creates a Namespace Manager with a fake clientset and the provided namespaces.
+func NewMockNamespaceManager(t *testing.T, enableDefaultGlobalNamespace bool, namespaces ...*slim_corev1.Namespace) Manager {
```

Comment:
No need to have this in a separate, non-test file, which also has the downside of including the `testing` dependency in a production package.